File: app/src/main/java/ua/com/foxminded/locationtrackera/model/bus/DefaultTrackerCache.java
package ua.com.foxminded.locationtrackera.model.bus;

import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.subjects.BehaviorSubject;

public class DefaultTrackerCache implements TrackerCache {

    private final BehaviorSubject<Boolean> serviceStatusSupplier = BehaviorSubject.create();

    @Override
    public void serviceStatusChanged(boolean isRunning) {
        serviceStatusSupplier.onNext(isRunning);
    }

    @Override
    public Observable<Boolean> setServiceStatusObservable() {
        return serviceStatusSupplier;
    }

}

File: models_impl/src/main/java/ua/com/foxminded/locationtrackera/models_impl/bus/DefaultTrackerCache.kt
package ua.com.foxminded.locationtrackera.models_impl.bus

import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.subjects.BehaviorSubject
import ua.com.foxminded.locationtrackera.models.bus.TrackerCache

class DefaultTrackerCache : TrackerCache {

    private val serviceStatusSupplier = BehaviorSubject.create<Boolean>()

    override fun serviceStatusChanged(isRunning: Boolean) {
        serviceStatusSupplier.onNext(isRunning)
    }

    override fun setServiceStatusObservable(): Observable<Boolean> = serviceStatusSupplier
}
